core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -231,6 +225,16 @@ private Object handleTxCommand(TxInvocationContext ctx, TransactionBoundaryComma
          }
       } else {
          if (currentTopologyId() > command.getTopologyId()) {
+            //in some cases, the node will not retry the transaction in this node and then it leaks.
+            //if it is safe, we rollback the transaction. If the coordinator retries on this node, no side effect is created.
```

Comment:
I'm only rollback when no locks are acquired, meaning that no updates will be made in this node (i.e. in the new topology, we are not an owner).
If we receive from state transfer, it will acquire the backups locks invalidating the rollback here.
If we receive the prepare before the state transfer, it will acquire locks too and it doesn't rollback.